### PR TITLE
PAINTROID-193 Implicit saving of edited image when returning to Catroid

### PR DESCRIPTION
**Pull request by FromG**
eliminated the dialog.
Removed the following tests:
testBrushToolBackPressedFromCatroidAndDiscardPicture
testBackToCatroidClickedWhenUndoAvailableAndOpenedFromCatroidThenShowSaveBeforeReturnDialog

Reason: The dialog no longer shows up which means we no longer need these tests.

### Your checklist for this pull request
Please review the [contributing guidelines](https://github.com/Catrobat/Paintroid/blob/develop/README.md) and [wiki pages](https://github.com/Catrobat/Catroid/wiki/) of this repository.

- [x] Include the name of the Jira ticket in the PR’s title
- [x] Include a summary of the changes plus the relevant context
- [x] Choose the proper base branch (*develop*)
- [x] Confirm that the changes follow the project’s coding guidelines
- [x] Verify that the changes generate no compiler or linter warnings
- [x] Perform a self-review of the changes
- [x] Verify to commit no other files than the intentionally changed ones
- [x] Include reasonable and readable tests verifying the added or changed behavior
- [x] Confirm that new and existing unit tests pass locally
- [x] Check that the commits’ message style matches the [project’s guideline](https://github.com/Catrobat/Catroid/wiki/Commit-Message-Guidelines)
- [x] Stick to the project’s gitflow workflow
- [x] Verify that your changes do not have any conflicts with the base branch
- [x] After the PR, verify that all CI checks have passed
- [ ] Post a message in the *#paintroid* [Slack channel](https://catrobat.slack.com) and ask for a code reviewer

### CODE REVIEW

**Review by Electronix1337 — CHANGES_REQUESTED** [review]
The two testfiles "OpenedFromPocketCodeWithImageTest" and "OpenedFromPocketCodeWithNewImageTest" are not proper written but I've already done that on Ticket 194 - Check all tests for correct behavior. I am going to adjust these tests when this ticket is merged into develop but I think you might want to change
`if (isImageUnchanged() || model.isSaved())` in MainActivityPresenter line 176 to
`if (isImageUnchanged() || model.isSaved() || model.isOpenedFromCatroid())`
 if I'm not mistaken so the save dialog is not shown when exiting the application when opened from catroid is true.

**Review by Electronix1337 — CHANGES_REQUESTED** [review]
testBrushToolBackPressedFromCatroidAndUsePicture in ToolOnBackPressedIntegrationTest.java can be deleted in my opinion because it tests the same as OpenedFromPocketCode and testBrushToolBackPressedFromCatroidAndDiscardPicture should be deleted because this is not possible anymore due to implicit saving. I would also suggest to adjust the OpenedFromPocketCodeNewImageTest.testSave because the save dialog should not show up and therefore the test has to be rewritten. In the current version the "openedfromcatroid" flag is not even set. And we should clarify if implicit saving from catroid should be done with png/jpg or if we let the user decide which would make the implicit saving kind of redundant? @wslany

**Review by wslany — CHANGES_REQUESTED** [review]
Currently this is difficult to test because it cannot be built with Catroid, see for instance in https://jenkins.catrob.at/job/Paintroid/job/PR-796/6/execution/node/43/log/ when you set the parameter yes for the build with Catroid in the build with parameter interface
```
07:27:37  FAILURE: Build failed with an exception.
07:27:37  
07:27:37  * What went wrong:
07:27:37  Execution failed for task ':catroid:processCatroidDebugManifest'.
07:27:37  > Manifest merger failed : Attribute application@name value=(org.catrobat.catroid.CatroidApplication) from AndroidManifest.xml:88:9-43
07:27:37    	is also present at [org.catrobat.paintroid:paintroid:2.5.2-LOCAL] AndroidManifest.xml:39:9-61 value=(androidx.multidex.MultiDexApplication).
07:27:37    	Suggestion: add 'tools:replace="android:name"' to <application> element at AndroidManifest.xml:34:5-44:19 to override.
07:27:37  
```
image

To run this, you need to be logged in in jenkins and press here:
image

**Review by wslany — CHANGES_REQUESTED** [review]
@FromG I just tried this with https://jenkins.catrob.at/job/Paintroid/job/PR-796/9/artifact/catroid/build/outputs/apk/catroid/debug/catroid-catroid-debug.apk : when I edit a look in a project, then return to the IDE part of Pocket Code, the changes are not there. I.e., the implicit saving is not done. All changes seem to be ignored. Also, when I try to add a new object and choose to draw it with Paintroid in Catroid, after going back out of Paintroid, there is no new object, all drawings are lost.

**Review by Electronix1337 — APPROVED** [review]
lgtm

**Review by wslany — APPROVED** [review]

**Review by wslany — CHANGES_REQUESTED** [review]
Perfect, thanks!! We can merge this PR once https://jira.catrob.at/browse/CATROID-761 has been implemented and merged, which is blocking the present ticket. We'll park the present ticket and PR in PO review until the Catroid ticket has been merged, hopefully soon. I mark the PR as "request change" so that nobody inadvertently merges it before the Catroid ticket has been merged.

**Review by wslany — APPROVED** [review]

**Comment by wslany** [comment]
Thanks!!

**Pull request merged**
